File: Assets/UnityChan2D/Demo/FSharpScripts/CoinController.fs
namespace UnityChan2D.Demo
open UnityEngine

type CoinController () =
  inherit MonoBehaviour()

  [<DefaultValue>] val mutable public getCoin : AudioClip

  member public this.OnTriggerEnter2D(other:Collider2D) =
    if (other.tag = "Player") then
      PointController.Instance.AddCoin()
      AudioSourceController.instance.PlayOneShot(this.getCoin)
      MonoBehaviour.Destroy(this.gameObject)

File: Assets/UnityChan2D/Demo/Scripts/CoinController.cs
using System.Collections;
using UnityEngine;
using UniRx;

public class CoinController : ObservableMonoBehaviour
{
    public AudioClip getCoin;

    public override void Awake()
    {
        this.OnTriggerEnter2DAsObservable()
            .Where(other => other.tag == "Player")
            .Subscribe(other =>
            {
                PointController.instance.AddCoin();
                AudioSourceController.instance.PlayOneShot(getCoin);
                Destroy(gameObject);
            });

        base.Awake();
    }
}
